### ISSUE TITLE
Validation service for Orlando documents not working

### ISSUE THREAD

**Issue by ilovan** [cwrc/CWRC-Writer#281]
@ajmacdonald , could you please see why we have this issue? I am in the process of evaluating older issues and this is essential for a number of the Orlando ones. 
Thanks,
Mihaela
![image](https://cloud.githubusercontent.com/assets/6600070/5807773/c21f07bc-9fe3-11e4-8198-ccc9074c977f.png)

**Comment by ajmacdonald** [comment]
There is a problem with the validation on the server.
I saw this error message:
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException
	org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:681)
	org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:585)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:641)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:76)

**Comment by brundin** [comment]
Jeff and I have been investigating this over the last while, and the issue
is being caused by the namespace reference at the top of the file, i.e.,
xmlns:cw="http://cwrc.ca/ns/cw#".  All four Orlando XML files - the Orlando
biography sample, the Orlando biography template, the Orlando writing
sample, and the Orlando writing template -- are failing because of the
namespace reference included in the root element.  Taking this namespace
reference out produces documents that validate.

The Orlando biography and writing schemas are well formed and valid, and
the four CWRC-Writer "template" XML files that are sent in the POST
requests to the CWRC Validator API validate in Oxygen.  There is something
strange going on in the Validator and its interaction with the Orlando
schemas when a namespace is included in the source document to be validated.

Can you remove the xmlns:cw="http://cwrc.ca/ns/cw#" references from the
POST request when validating Orlando documents?  That seems to be the
strange workaround that will allow us to validate Orlando documents using
the CWRC Validator API.

Mike and Jeff

**Comment by brundin** [comment]
Jeff and I altered the Java code in the CWRC XML Validator API to provide more informative error diagnostics.  Based on what we learned from the additional error information that was produced, we focused on how the root elements are constructed in all of the schemas.  We learned that there is a funny "glitch" in the Validator that requires the root element of a schema to have an optional attribute defined in order for an xmlns:namespace attribute in the root element to validate correctly.  Optional @ID attributes were added to the BIOGRAPHY root element (in orlando_biography.rng) and the WRITING root element (orlando_writing.rng), so the CWRC-Writer Orlando templates now validate correctly against these schemas when they have the xmlns:cw="http://cwrc.ca/ns/cw#" attribute namespace added to the BIOGRAPHY and WRITING root elements.  -- MRB & JCA
